app/models/host_info_providers/puppet_info.rb

```diff
@@ -21,8 +21,23 @@ def puppetclass_parameters
       values = keys.values_hash(host)
 
       klasses = {}
+      klass_error = {}
       host.classes.each do |klass|
-        klasses[klass.name] = smart_class_params_for(klass, key_hash, values)
+        begin
+          klasses[klass.name] = smart_class_params_for(klass, key_hash, values)
+        rescue StandardError => exc
+          klass_error[klass] = exc
+        end
+      end
+      if klass_error.size
+        # FIXME: Is logger in scope under another name?
+        logger = Foreman::Logging.logger('app')
```

Comment:
App logger should be available through `logger` method in this context I believe.